### PR TITLE
Add Avro file format product tests

### PR DESCRIPTION
**Pull request by sudarsanm94**
AVRO file format doesn't support tinyint and smallint datatypes.
AVRO file format doesn't support coercion from bigint to string datatype.

### CODE REVIEW

File: presto-product-tests/src/main/java/com/facebook/presto/tests/hive/AllSimpleTypesTableDefinitions.java [hunk 1]
```diff
@@ -54,6 +51,10 @@ private AllSimpleTypesTableDefinitions()
             .setNoData()
             .build();
 
+    private AllSimpleTypesTableDefinitions()
```
Comment (rschlussel) [inline presto-product-tests/src/main/java/com/facebook/presto/tests/hive/AllSimpleTypesTableDefinitions.java]: Move this to the top of the class.

File: presto-product-tests/src/main/java/com/facebook/presto/tests/hive/TestHiveStorageFormats.java [hunk 2]
```diff
@@ -53,10 +53,54 @@
                 {storageFormat("RCBINARY")},
                 {storageFormat("RCTEXT")},
                 {storageFormat("SEQUENCEFILE")},
-                {storageFormat("TEXTFILE")}
+                {storageFormat("TEXTFILE")},
+                {storageFormat("AVRO")}
         };
     }
 
+    private static void assertSelect(String query, String tableName)
```
Comment (rschlussel) [inline presto-product-tests/src/main/java/com/facebook/presto/tests/hive/TestHiveStorageFormats.java]: Unrelated change.  Move this back to where it was.

File: presto-product-tests/src/main/java/com/facebook/presto/tests/hive/TestHiveCoercion.java [hunk 3]
```diff
@@ -206,16 +218,52 @@ public void testHiveCoercionParquet()
...
             throws SQLException
     {
         String tableName = mutableTableInstanceOf(tableDefinition).getNameInDatabase();
 
         executeHiveQuery(format("INSERT INTO TABLE %s " +
-                "PARTITION (id=1) " +
-                "VALUES" +
-                "(-1, 2, -3, 100, -101, 2323, 12345, 0.5)," +
-                "(1, -2, null, -100, 101, -2323, -12345, -1.5)",
+                        "PARTITION (id=1) " +
```
Comment (rschlussel) [inline presto-product-tests/src/main/java/com/facebook/presto/tests/hive/TestHiveCoercion.java]: undo formatting change

File: presto-product-tests/src/main/java/com/facebook/presto/tests/hive/TestHiveCoercion.java [hunk 4]
```diff
@@ -56,24 +56,22 @@
...
-
-    public static final HiveTableDefinition HIVE_COERCION_PARQUET = parquetTableDefinitionBuilder()
+    public static final HiveTableDefinition HIVE_COERCION_AVRO = avroTableDefinitionBuilder()
+            .setNoData()
+            .build();
+    private static String tableNameFormat = "%s_hive_coercion";
+    public static final HiveTableDefinition HIVE_COERCION_TEXTFILE = tableDefinitionBuilder("TEXTFILE", Optional.empty(), Optional.of("DELIMITED FIELDS TERMINATED BY '|'"))
             .setNoData()
             .build();
-
```
Comment (rschlussel) [inline presto-product-tests/src/main/java/com/facebook/presto/tests/hive/TestHiveCoercion.java]: undo line deletion here and in the rest of the file

File: presto-product-tests/src/main/java/com/facebook/presto/tests/hive/TestHiveCoercion.java [hunk 5]
```diff
@@ -56,24 +56,22 @@
 public class TestHiveCoercion
         extends ProductTest
 {
-    private static String tableNameFormat = "%s_hive_coercion";
```
Comment (rschlussel) [inline presto-product-tests/src/main/java/com/facebook/presto/tests/hive/TestHiveCoercion.java]: you've reordered things in this file.  Keep the order and formatting as it was.

File: presto-product-tests/src/main/java/com/facebook/presto/tests/hive/TestTablePartitioningSelect.java [hunk 6]
```diff
@@ -57,6 +58,7 @@ private static HiveTableDefinition singleIntColumnPartitionedTableDefinition(Str
         String tableName = fileFormat.toLowerCase() + "_single_int_column_partitioned";
         HiveDataSource dataSource = createResourceDataSource(tableName, String.valueOf(ThreadLocalRandom.current().nextLong(Long.MAX_VALUE)), "com/facebook/presto/tests/hive/data/single_int_column/data." + fileFormat.toLowerCase());
         HiveDataSource invalidData = createStringDataSource(tableName, String.valueOf(ThreadLocalRandom.current().nextLong(Long.MAX_VALUE)), "INVALID DATA");
+
```
Comment (rschlussel) [inline presto-product-tests/src/main/java/com/facebook/presto/tests/hive/TestTablePartitioningSelect.java]: undo line add

File: presto-product-tests/src/main/java/com/facebook/presto/tests/hive/TestAllDatatypesFromHiveConnector.java [hunk 7]
```diff
@@ -73,48 +74,33 @@
 public class TestAllDatatypesFromHiveConnector
         extends ProductTest
 {
-    public static final class TextRequirements
-            implements RequirementsProvider
+    private static TableInstance mutableTableInstanceOf(TableDefinition tableDefinition)
```
Comment (rschlussel) [inline presto-product-tests/src/main/java/com/facebook/presto/tests/hive/TestAllDatatypesFromHiveConnector.java]: Undo the reordering.

**Comment by sudarsanm94** [comment]
Made the requested changes

File: presto-product-tests/src/main/java/com/facebook/presto/tests/hive/TestHiveCoercion.java [hunk 8]
```diff
@@ -116,6 +120,24 @@
                         "STORED AS PARQUET");
     }
 
+    private static HiveTableDefinition.HiveTableDefinitionBuilder avroTableDefinitionBuilder()
+    {
+        return HiveTableDefinition.builder("avro_hive_coercion")
+                .setCreateTableDDLTemplate("" +
+                        "CREATE TABLE %NAME%(" +
+                        "    int_to_bigint              INT," +
```
Comment (petroav) [inline presto-product-tests/src/main/java/com/facebook/presto/tests/hive/TestHiveCoercion.java]: Looks like Avro is pretty limited in terms of coercions. We should create another patch that documents these limitations? What do you think @rschlussel?

File: presto-product-tests/src/main/java/com/facebook/presto/tests/hive/TestHiveCoercion.java [hunk 9]
```diff
@@ -206,6 +238,42 @@ public void testHiveCoercionParquet()
...
 
+    @Requires(AvroRequirements.class)
+    @Test(groups = {HIVE_COERCION, HIVE_CONNECTOR})
+    public void testHiveCoercionAvro()
+            throws SQLException
+    {
+        doAvroTestHiveCoercion(HIVE_COERCION_AVRO);
+    }
+
+    private void doAvroTestHiveCoercion(HiveTableDefinition tableDefinition)
```
Comment (petroav) [inline presto-product-tests/src/main/java/com/facebook/presto/tests/hive/TestHiveCoercion.java]: Inline this method in the test. It makes sense to abstract the logic of `doTestHiveCoercion` because it is being called from multiple tests but this logic is only called from the Avro test.

File: presto-product-tests/src/main/java/com/facebook/presto/tests/hive/TestAllDatatypesFromHiveConnector.java [hunk 10]
```diff
@@ -385,4 +486,16 @@ private static TableHandle tableHandleInSchema(TableDefinition tableDefinition)
         }
         return tableHandle;
     }
+
+    public static final class AvroRequirements
```
Comment (petroav) [inline presto-product-tests/src/main/java/com/facebook/presto/tests/hive/TestAllDatatypesFromHiveConnector.java]: This should be located at the top of the class next to the file type requirements.

File: presto-product-tests/src/main/java/com/facebook/presto/tests/hive/TestHiveCoercion.java [hunk 11]
```diff
@@ -116,6 +120,24 @@
...
+                .setCreateTableDDLTemplate("" +
+                        "CREATE TABLE %NAME%(" +
+                        "    int_to_bigint              INT," +
+                        "    float_to_double            DOUBLE" +
+                        ") " +
+                        "PARTITIONED BY (id BIGINT) " +
+                        "STORED AS AVRO");
+    }
+
+    private static void avroAlterTableColumnTypes(String tableName)
```
Comment (petroav) [inline presto-product-tests/src/main/java/com/facebook/presto/tests/hive/TestHiveCoercion.java]: This method should be located next to `alterTableColumnTypes()`.

File: presto-product-tests/src/main/java/com/facebook/presto/tests/hive/TestHiveCoercion.java [hunk 12]
```diff
@@ -307,6 +384,21 @@ private static void alterTableColumnTypes(String tableName)
         executeHiveQuery(format("ALTER TABLE %s CHANGE COLUMN float_to_double float_to_double double", tableName));
     }
 
+    private void avroAssertColumnTypes(QueryResult queryResult)
```
Comment (petroav) [inline presto-product-tests/src/main/java/com/facebook/presto/tests/hive/TestHiveCoercion.java]: This method should be moved to be next to `assertColumnTypes()`.

**Comment by petroav** [comment]
A couple of minor comments. Also, can you re-write your commit message to follow this format https://chris.beams.io/posts/git-commit/?

I would write something like this: "Add Avro file format product tests".

**Comment by sudarsanm94** [comment]
I have made all the suggested changes.

**Comment by anusudarsan** [comment]
superseded by https://github.com/prestodb/presto/pull/9624
